.buildkite/hooks/pre-command

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -euo pipefail
+
+DOCKER_REGISTRY_SECRET_PATH="kv/ci-shared/platform-ingest/docker_registry_prod"
+
+if [[ "$BUILDKITE_PIPELINE_SLUG" == "temp-fleet-server" && "$BUILDKITE_STEP_KEY" == "publish" ]]; then #should be changed to "fleet-server" after tests
+  export DOCKER_USER=$(retry 5 vault kv get -field user "${DOCKER_REGISTRY_SECRET_PATH}")
+  export DOCKER_USER_SECRET=$(retry 5 vault kv get -field password "${DOCKER_REGISTRY_SECRET_PATH}")
```

Comment:
Just to be in the safe side, I would follow as in Jenkins and set as redacted both environment variables here. Even if they are just used here in the pre-command as suggested in https://github.com/elastic/fleet-server/pull/2604/files#r1197548706

Let's avoid any possible issue with those having them as redacted vars from the beginning.

```suggestion
  export DOCKER_USERNAME_SECRET=$(retry 5 vault kv get -field user "${DOCKER_REGISTRY_SECRET_PATH}")
  export DOCKER_PASSWORD_SECRET=$(retry 5 vault kv get -field password "${DOCKER_REGISTRY_SECRET_PATH}")
```

`_SECRET` is a valid suffix for redacted-vars:
https://buildkite.com/docs/pipelines/managing-log-output#redacted-environment-variables

Reply:
fixed